scripts/imgtool/image.py

```diff
@@ -360,26 +373,37 @@ def add_header(self, enckey, protected_tlv_size):
         self.payload = bytearray(self.payload)
         self.payload[:len(header)] = header
 
-    def _trailer_size(self, write_size, max_sectors, overwrite_only, enckey):
+    def _trailer_size(self, write_size, max_sectors, overwrite_only, enckey,
+                      save_enctlv, minimal_enctlv):
         # NOTE: should already be checked by the argument parser
         magic_size = 16
         if overwrite_only:
             return MAX_ALIGN * 2 + magic_size
         else:
             if write_size not in set([1, 2, 4, 8]):
-                raise Exception("Invalid alignment: {}".format(write_size))
+                raise click.BadParameter("Invalid alignment: {}".format(
+                    write_size))
             m = DEFAULT_MAX_SECTORS if max_sectors is None else max_sectors
             trailer = m * 3 * write_size  # status area
             if enckey is not None:
-                trailer += 16 * 2  # encryption keys
+                if save_enctlv:
+                    try:
+                        keylen = len(enckey.get_private_bytes(minimal_enctlv))
```

Comment:
This actually needs to be rounded to `align` size to avoid the bootloader possibly copying some non-aligned size (if image data overlaps the last sector containing the trailer).